### PR TITLE
Some docs/exception messages for bulk delete

### PR DESCRIPTION
**Pull request by roji**

### CODE REVIEW

File: src/EFCore.Relational/Extensions/RelationalQueryableExtensions.cs [hunk 1]
```diff
@@ -236,19 +236,31 @@ public static DbCommand CreateDbCommand(this IQueryable source)
     #region BulkDelete
 
     /// <summary>
-    ///     TBD
+    ///     Deletes all entity instances which match the LINQ query from the database.
     /// </summary>
+    /// <remarks>
```
Comment (smitpatel) [inline src/EFCore.Relational/Extensions/RelationalQueryableExtensions.cs:241]: Copy remarks to async thing also

File: src/EFCore.Relational/Query/RelationalQueryableMethodTranslatingExpressionVisitor.cs [hunk 2]
```diff
@@ -1042,6 +1042,10 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCallExp
     /// <summary>
     ///     Validates if the current select expression can be used for bulk delete operation or it requires to be pushed into a subquery.
     /// </summary>
+    /// <remarks>
+    ///     By default, only single-table select expressions are supported, and only with a predicate.
+    ///     Providers can override this to allow more select expression features to be supported without pushing down into a subquery.
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalQueryableMethodTranslatingExpressionVisitor.cs:1047]: Also add remark that if they support more here, they also need to update Sql Gen

File: src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.cs [hunk 3]
```diff
@@ -49,10 +49,11 @@ protected override Expression VisitExtension(Expression extensionExpression)
             : base.VisitExtension(extensionExpression);
 
     /// <summary>
-    ///  asdagv
+    ///     Visits the given <paramref name="nonQueryExpression" />, returning an expression that when compiled, can execute the operation
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.cs:52]:
```suggestion
    ///     Visits the given <paramref name="nonQueryExpression" />, returning an expression that when compiled, can execute the non query operation
```

File: src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.cs [hunk 4]
```diff
@@ -49,10 +49,11 @@ protected override Expression VisitExtension(Expression extensionExpression)
...
+    ///     against the database.
     /// </summary>
-    /// <param name="nonQueryExpression">asdasfdsa</param>
-    /// <returns>asdasd</returns>
+    /// <param name="nonQueryExpression">The expression to be compiled.</param>
+    /// <returns>An expression ready to be compiled and executed.</returns>
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.cs:56]: An expression which executes non query operation.

**Comment by smitpatel** [comment]
Thanks!

**Review by smitpatel — APPROVED** [review]

**Comment by roji** [comment]
(feel free to merge at all, it's only a PR branch)

**Pull request merged**